packages/account/src/BiconomySmartAccountV2.ts

```diff
@@ -272,21 +276,31 @@ export class BiconomySmartAccountV2 extends BaseSmartAccount {
     super.validateUserOp(userOp, requiredFields)
     const userOpHash = await this.getUserOpHash(userOp)
 
-    // Note: we could use signUserOp from validationModule
-    const sig = await this.activeValidationModule.signMessage(arrayify(userOpHash))
+    const sig = await this.activeValidationModule.signUserOpHash(userOpHash, moduleSignerInfo)
```

Comment:
Didn't understand this change. The decoding of signature where module address and rest of the data is decoded, happens in Smart Account on chain. So this responsibility also should be with BiconomySmartAccount class in the SDK. 
Tomorrow if we upgrade the Smart Account which has new decoding logic, then we would need to change the logic in all validation modules in the SDK as well, if the module address is appended by validation modules.

Reply:
do you mean for this specific line or juts the fact that signatureWithModuleAddress should be created in SA?

Reply:
Yeah i meant in general, for all validation modules.. the module address should be appended in SA class.

Reply:
ok will make changes.

Reply:
done